axelrod/strategies/punisher.py

```diff
@@ -119,3 +119,42 @@ def reset(self):
         self.grudged = False
         self.grudge_memory = 0
         self.mem_length = 1
+
+class LevelPunisher(Player):
+    """
+    Names:
+
+    - Level Punisher: Original by Geraint Palmer
+    """
+
+    name = 'Level Punisher'
+    classifier = {
+        'memory_depth': float('inf'),  # Long memory
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        super(LevelPunisher, self).__init__()
+        self.history = []
+        self.mem_length = 1
+        self.grudged = False
+        self.grudge_memory = 1
+        
+    def strategy(self, opponent):
+        
+        if self.grudge_memory >= self.mem_length:
+            self.grudge_memory = 0
+            self.grudged = False
```

Comment:
@souravsingh this is why the test is failing. The `strategy` method needs to return a `C` or a `D`. At the moment for example the following test is failing because this method returns `None` (nothing):

```
======================================================================
FAIL: test_strategy (axelrod.tests.unit.test_punisher.TestLevelPunisher)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vince/src/Axelrod/axelrod/tests/unit/test_punisher.py", line 140, in test_strategy
    self.responses_test([], [], [C], attrs={"grudged": False})
  File "/home/vince/src/Axelrod/axelrod/tests/unit/test_player.py", line 285, in responses_test
    random_seed=random_seed, attrs=attrs)
  File "/home/vince/src/Axelrod/axelrod/tests/unit/test_player.py", line 139, in test_responses
    test_class.assertEqual(s1, response)
AssertionError: None != 'C'
```

Take a look at the `strategy` method for the original `Punisher` strategy: https://github.com/Axelrod-Python/Axelrod/blob/master/axelrod/strategies/punisher.py#L37

I hope that helps.